### ISSUE TITLE
[UWP] selectAction support not working on latest master commit

### ISSUE THREAD

**Issue by BenAffleckIsBatman** [Microsoft/AdaptiveCards#1572]
Hi - we realized a bit too late that selectAction is not supported in the UWP SDK as of now, so I searched for PRs and found #1396 fully merged into master. 

I forked master, created a nuget package my own and replaced it with the official AdaptiverCards.Renderer. Unfortunately the renderer still not renders selectAction and simply returns null. 

Here you find a little sample causing this. Please note that when removing selectAction from the JSON, it all works fine:

```
{
    {
        "type": "AdaptiveCard",
        "version": "1.0",
        "body": [{
            "type": "TextBlock",
            "size": "large",
            "weight": "bolder",
            "text": "Who would you like to invite?"
        }, {
            "type": "TextBlock",
            "text": "Please give me the E-Mail of this Person."
        }, {
            "type": "TextBlock",
            "text": "Your team leader or your team:"
        }, {
            "type": "ColumnSet",
            "columns": [{
                "type": "Column",
                "width": "15",
                "items": [{
                    "type": "Image",
                    "url": "https://i.imgur.com/5F63et4.png"
                }],
                "selectAction": {
                    "type": "Action.Submit",
                    "data": "team.leader@mail.ch"
                }
            }, {
                "type": "Column",
                "width": "85",
                "items": [{
                    "type": "TextBlock",
                    "weight": "bolder",
                    "text": "Teamleader Name",
                    "spacing": "none",
                    "separation": "none"
                }, {
                    "type": "TextBlock",
                    "text": "team.leader@mail.ch",
                    "spacing": "none",
                    "separation": "none"
                }],
                "selectAction": {
                    "type": "Action.Submit",
                    "data": "team.leader@mail.ch"
                }
            }]
        }, {
            "type": "ColumnSet",
            "columns": [{
                "type": "Column",
                "width": "15",
                "items": [{
                    "type": "Image",
                    "url": "https://i.imgur.com/5F63et4.png"
                }],
                "selectAction": {
                    "type": "Action.Submit",
                    "data": "your.team@mail.ch"
                }
            }, {
                "type": "Column",
                "width": "85",
                "items": [{
                    "type": "TextBlock",
                    "weight": "bolder",
                    "text": "Your Team Name",
                    "spacing": "none",
                    "separation": "none"
                }, {
                    "type": "TextBlock",
                    "text": "your.team@mail.ch",
                    "spacing": "none",
                    "separation": "none"
                }],
                "selectAction": {
                    "type": "Action.Submit",
                    "data": "your.team@mail.ch"
                }
            }]
        }, {
            "type": "TextBlock",
            "text": "Recent contacts:"
        }, {
            "type": "ColumnSet",
            "columns": [{
                "type": "Column",
                "width": "15",
                "items": [{
                    "type": "Image",
                    "url": "https://i.imgur.com/5F63et4.png"
                }],
                "selectAction": {
                    "type": "Action.Submit",
                    "data": "alena.kennedy@mail.ch"
                }
            }, {
                "type": "Column",
                "width": "85",
                "items": [{
                    "type": "TextBlock",
                    "weight": "bolder",
                    "text": "Alena Kennedy",
                    "spacing": "none",
                    "separation": "none"
                }, {
                    "type": "TextBlock",
                    "text": "alena.kennedy@mail.ch",
                    "spacing": "none",
                    "separation": "none"
                }],
                "selectAction": {
                    "type": "Action.Submit",
                    "data": "alena.kennedy@mail.ch"
                }
            }]
        }, {
            "type": "ColumnSet",
            "columns": [{
                "type": "Column",
                "width": "15",
                "items": [{
                    "type": "Image",
                    "url": "https://i.imgur.com/5F63et4.png"
                }],
                "selectAction": {
                    "type": "Action.Submit",
                    "data": "sean.ammann@mail.ch"
                }
            }, {
                "type": "Column",
                "width": "85",
                "items": [{
                    "type": "TextBlock",
                    "weight": "bolder",
                    "text": "Sean Ammann",
                    "spacing": "none",
                    "separation": "none"
                }, {
                    "type": "TextBlock",
                    "text": "sean.ammann@mail.ch",
                    "spacing": "none",
                    "separation": "none"
                }],
                "selectAction": {
                    "type": "Action.Submit",
                    "data": "sean.ammann@mail.ch"
                }
            }]
        }]
    }
}
```

**Comment by BenAffleckIsBatman** [comment]
@tuanh118 Any chance to have a look at this? I think this is really a show stopper for many scenarios. Would really appreciate your help with this. Thanks!

**Comment by anbare** [comment]
Hey @BenAffleckIsBatman, thanks for using Cards! selectAction is supported by UWP.

However, **you must include a title property** for the action to be a valid action, as **title** is a required property per the [Card schema](http://adaptivecards.io/explorer/Action.Submit.html).

I've created an issue to update the web visualizer so that it fails when a title isn't provided (#1604), since today your payload works perfectly in the web visualizer, even though it shouldn't.

I'll close this issue. If it's still not working, re-open it, or if you have any other questions, create a new issue!

**Comment by BenAffleckIsBatman** [comment]
Thanks @anbare ! 

I added the title property and it works fine in UWP renderer. 

Could you please explain the following to me:

1. Did this support come with #1396 ? If so, why is it part of Milestone 1806 lable while still being available through the latest nuget 1.0.1 published April 2018 

2. This doc is telling me that selectAction is not supported in UWP. https://docs.microsoft.com/en-us/adaptive-cards/display/rendererstatus - is this doc not up to date then ? 

3. Wouldn't it be better to throw a unspecific/overall schema validation exception instead of returning just null as long as no proper validation is implemented on the platform SDKs. This would have saved me a lot of time as I searched way into the wrong direction

Thanks for your help!

**Comment by anbare** [comment]
Thanks!

1. No, selectAction for columns and whatnot came way earlier than #1396. #1396 was for the top-level selectAction (a click action for the **entire** card), which was a new 1.1 feature.
2. Yes, seems like that doc isn't up-to-date, I'll look at taking a full look over that doc and updating everything.
3. Instead of throwing exceptions, we return errors and warnings in the RenderResult. If you inspect the Errors property on the RenderResult using the latest master, you'll see that it returns a `Property is required but was found empty: title` error on your payload.

I strongly recommend outputting those errors/warnings when developing, so that you know why things are failing!
